api/v1/client/daemon/daemon_client.go

```diff
@@ -211,6 +213,46 @@ func (a *Client) GetDebuginfo(params *GetDebuginfoParams, opts ...ClientOption)
 	panic(msg)
 }
 
+/*
+GetHealth gets modules health of cilium daemon
+
+Returns modules health and status information of the Cilium daemon.
+*/
+func (a *Client) GetHealth(params *GetHealthParams, opts ...ClientOption) (*GetHealthOK, error) {
+	// TODO: Validate the params before sending
+	if params == nil {
+		params = NewGetHealthParams()
+	}
+	op := &runtime.ClientOperation{
+		ID:                 "GetHealth",
+		Method:             "GET",
+		PathPattern:        "/health",
+		ProducesMediaTypes: []string{"application/json"},
+		ConsumesMediaTypes: []string{"application/json"},
+		Schemes:            []string{"http"},
+		Params:             params,
+		Reader:             &GetHealthReader{formats: a.formats},
+		Context:            params.Context,
+		Client:             params.HTTPClient,
+	}
+	for _, opt := range opts {
+		opt(op)
+	}
+
+	result, err := a.transport.Submit(op)
+	if err != nil {
+		return nil, err
+	}
+	success, ok := result.(*GetHealthOK)
+	if ok {
+		return success, nil
+	}
+	// unexpected success response
+	// safeguard: normally, absent a default response, unknown success responses return an error above: so this is a codegen issue
+	msg := fmt.Sprintf("unexpected success response for GetHealth: API contract not enforced by server. Client expected to get an error, but got: %T", result)
```

Comment:
Not sure if this would be that useful here given we are expecting a specific type that somehow did not come across the wire. Imho this is a clearer error that dumping the type and entire resp content which would not be that actionable?

Reply:
You're right, I agree that %T is better.